### ISSUE TITLE
Latest version in chrome extension webstore (0.9.21) breaks multiple features

### ISSUE THREAD

**Issue by deadcyclo** [brookhong/Surfingkeys#607]
Omnibar is not available in the configfile/Front context which leads to multiple issues (including breaking documented features):

1. Advanced search engines added with addSearchAliasX that use Omnibar.listResults to show custom results are broken.
2. The documented features in the readme "Omnibar helpers" (Omnibar.listWords(<array of words>)
Omnibar.html(<any html snippets>)) no longer work
3. cmap and cmapkeys functions no longer work, since they check that Omnibar !=== 'undefined', so no custom keybindings in the omnibar any more.
4. Listsessions no longer works due to it checking !Front.omnibar.is(":visible")

Basically, we need Omnibar back at a global level.

Also, why is the version number ahead of the version number here?

**Comment by brookhong** [comment]
Thanks for reporting this.

In short, this issue together with #601 were introduced by the change of 0.9.19, which removed `unsafe-eval` from content security policy.

0.9.21 was released to fix #601, I will push the commit soon. 

Due to the removal of `unsafe-eval`, user scripts could not be executed in context of the extension itself, which means Omnibar is not accessible in user space now.

For addSearchAliasX, please don't use Omnibar.listResults, just return an array, you could refer to the default mappings. Omnibar.html may be removed, since it is unsafe.

**Comment by deadcyclo** [comment]
Indeed. I was afraid it was that.

For most searches, I do return an array. However I have some searches where url+title isn't enough. A concrete example is the employee search I've made. Searches our solr installation and shows name, title, telephone, email and a picture (and links to their page on the intranet). Unfortunately, that doesn't work any more.

Would you consider adding a feature to `self.createURLItem = function(b, rxp) { ` that accepted an array of `{ url: 'xxx', html: '.....' }`? So basically we could return an array of url and content of the div to be rendered? This would allow us to keep our advanced omnibar search usage, without unsafe-eval.

**Comment by Jamie-505** [comment]
```
mapkey('J', 'switch to left tab', 'RUNTIME("previousTab")');
mapkey('K', 'switch to right tab', 'RUNTIME("nextTab")');
```
also ist not working anymore

**Comment by brookhong** [comment]
This has been mentioned in [changelog](https://github.com/brookhong/Surfingkeys/blob/master/pages/changelog.md).

**Comment by Jamie-505** [comment]
@brookhong oh sry missed it 😅

**Comment by deadcyclo** [comment]
@brookhong Cool. I'll try to get on to it soon. And yeah, parsing html strings isn't something that really should be done unless needed. I see my example was kind of misleading. IMHO the only sensible way to do it is to accept a dom-node, so I guess the key 'html' probably should be 'node' to not be misleading.

**Comment by petobens** [comment]
@brookhong I'm trying to add a custom search engine for github. I used to use the one [here](https://gist.github.com/b0o/a91d5fca1d762ecf43ea75af3685d651#file-surfingkeys-js-L221).  Since omnibar is no longer accesible currently I'm using:
```javascript
addSearchAliasX('h', 'GitHub',
    'https://github.com/search?q=',
    's',
    ('https://api.github.com/search/repositories?order=desc&q='),
    function(response) {
        var res = JSON.parse(response.text)['items'];
        return res.map(function(r){
            return r.name;
        });
    });
```
however this takes me to github search rather than to the selected candidate repo:
![may-25-2018 16-23-38](https://user-images.githubusercontent.com/2583971/40562711-8b0c4b04-6038-11e8-8713-3c8cc92c8ec8.gif)


Any ideas on how to fix it?

**Comment by brookhong** [comment]
Please use below:

    addSearchAliasX('h', 'GitHub',
        'https://github.com/search?q=',
        's',
        ('https://api.github.com/search/repositories?order=desc&q='),
        function(response) {
            var res = JSON.parse(response.text)['items'];
            return res.map(function(r){
                return {
                    title: r.description,
                    url: r.html_url
                };
            });
        });

**Comment by petobens** [comment]
Thank you @brookhong ! That indeed works. One minor question though, I've noticed (as in the GIF) that when i use a full window, then the omnibar prompt disappears when the results are listed and in order for it to re appear I need to press some key (such as space):

![may-27-2018 11-50-31](https://user-images.githubusercontent.com/2583971/40587332-4006fa4c-61a4-11e8-80ff-fea1a62120a5.gif)

Can this be fixed?
